src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -89,132 +99,155 @@ public function handle(AddProductToOrderCommand $command)
     {
         $order = $this->getOrderObject($command->getOrderId());
 
-        if ($this->context->cart !== null) {
-            $oldCartRules = $this->context->cart->getCartRules();
-        } else {
-            $oldCartRules = [];
-        }
-
-        $this->assertOrderWasNotShipped($order);
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
+            ->setCustomer(new Customer($order->id_customer));
```

Comment:
Thanks to this I think `$this->context->customer = new Customer($order->id_customer);` L328 is now useless

Reply:
:+1: